spinoso-string/src/enc/ascii/mod.rs

```diff
@@ -343,6 +343,26 @@ impl AsciiString {
     }
 }
 
+// Index
+impl AsciiString {
+    #[inline]
+    #[must_use]
+    pub fn index(&self, needle: &[u8], offset: usize) -> Option<usize> {
+        let buf = self.get(offset..)?;
+        let index = buf.find(needle)?;
+        Some(index + offset)
+    }
+
+    #[inline]
+    #[must_use]
+    pub fn rindex(&self, needle: &[u8], offset: usize) -> Option<usize> {
+        let search_until = (offset + 1).min(self.len());
+        let buf = self.get(..search_until)?;
```

Comment:
this can be simplified to avoid a branch. combined with the above overflow comment, I think this works:

```suggestion
        let buf = self.get(..=search_until).unwrap_or_else(|| self.get(..));
```

Reply:
this suggestion might need a special case and `#[test]` for when `buf` is empty.

Reply:
```console
[3.2.2] > "".rindex ""
=> 0
[3.2.2] > "".rindex "a"
=> nil
[3.2.2] > "a".rindex ""
=> 1
```

Reply:
I added some tests in https://github.com/artichoke/artichoke/pull/2554 which caught a few underflows.